hathor/ipykernel.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2024 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from logging import getLogger
+from typing import TYPE_CHECKING, Any, Optional
+
+from ipykernel.kernelapp import IPKernelApp as OriginalIPKernelApp
+
+if TYPE_CHECKING:
+    from hathor.manager import HathorManager
+
+
+class IPKernelApp(OriginalIPKernelApp):
+    def __init__(self, runtime_dir: Optional[str] = None):
+        super().__init__()
+        if runtime_dir is not None:
+            self.connection_dir = runtime_dir
+        self.connection_file = 'kernel.json'
+        self.logging_config: dict[str, Any] = {}  # empty out logging config
+        self.default_extensions: list[str] = []  # disable default extensions
+        self.no_stderr = True  # disable forwarding of stderr (because we use it for logging)
+        self.log = getLogger('hathor.ipykernel')
+
+    # ignore registering of signals
+    def init_signal(self) -> None:
+        pass
+
+    # custom logging of connection file
+    def log_connection_info(self) -> None:
+        self.log.info(f'ipykernel connection enabled at {self.abs_connection_file}')
+
+    # disable original logging setup
+    def get_default_logging_config(self) -> dict[str, Any]:
+        return {"version": 1, "disable_existing_loggers": False}
+
+    def configure_tornado_logger(self) -> None:
+        pass
```

Comment:
It's not that we are not using the tornado logger. This method will call `logger.dictConfig` with some settings that will basically result in it logging with a different format, but we already have our own logger config. Overriding this method is just to prevent our logging config from breaking.

Reply:
Thanks!